File: PowerUp.ts
import { Sprite } from './Sprite';
import { Player } from './Player';
import { GamePanel } from './GamePanel';

export abstract class PowerUp extends Sprite {
    constructor(img: HTMLImageElement, x: number, y: number) {
        super(img, x, y);
    }
    abstract applyEffect(player: Player, panel: GamePanel): void;
}

File: PowerUp.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package com.mycompany.mazegame;

/**
 *
 * @author lsuti
 */
import java.awt.image.BufferedImage;

public abstract class PowerUp extends Sprite {

    public PowerUp(BufferedImage img, int x, int y) {
        super(img, x, y);
    }

    public abstract void applyEffect(Player player, GamePanel panel);
}
